clients/redigo.go

```diff
@@ -0,0 +1,389 @@
+
+	if len(keys) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	for _, key := range keys {
+		args = append(args, key)
+	}
+
+	args = append(args, path)
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_MGET, args...)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONDel to delete a json object
+//
+// ReJSON syntax:
+// 	JSON.DEL <key> <path>
+//
+func (r *Redigo) JSONDel(key string, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_DEL, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONType to get the type of key or member at path.
+//
+// ReJSON syntax:
+// 	JSON.TYPE <key> [path]
+//
+func (r *Redigo) JSONType(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_TYPE, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONNumIncrBy to increment a number by provided amount
+//
+// ReJSON syntax:
+// 	JSON.NUMINCRBY <key> <path> <number>
+//
+func (r *Redigo) JSONNumIncrBy(key, path string, number int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_NUMINCRBY, key, path, number)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONNumMultBy to increment a number by provided amount
+//
+// ReJSON syntax:
+// 	JSON.NUMMULTBY <key> <path> <number>
+//
+func (r *Redigo) JSONNumMultBy(key, path string, number int) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_NUMMULTBY, key, path, number)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONStrAppend to append a jsonstring to an existing member
+//
+// ReJSON syntax:
+// 	JSON.STRAPPEND <key> [path] <json-string>
+//
+func (r *Redigo) JSONStrAppend(key, path, jsonstring string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_STRAPPEND, key, path, jsonstring)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONStrLen to return the length of a string member
+//
+// ReJSON syntax:
+// 	JSON.STRLEN <key> [path]
+//
+func (r *Redigo) JSONStrLen(key, path string) (res interface{}, err error) {
+
+	name, args, err := rjs.CommandBuilder(rjs.ReJSONCommand_STRLEN, key, path)
+	if err != nil {
+		return nil, err
+	}
+	return r.Conn.Do(name, args...)
+}
+
+// JSONArrAppend to append json value into array at path
+//
+// ReJSON syntax:
+// 	JSON.ARRAPPEND <key> <path> <json> [json ...]
+//
+func (r *Redigo) JSONArrAppend(key, path string, values ...interface{}) (res interface{}, err error) {
+
+	if len(values) == 0 {
+		return nil, rjs.NeedAtleastOneArg
+	}
+	args := make([]interface{}, 0)
+	args = append(args, key, path)
+	args = append(args, values...)
```

Comment:
Combine the two appends.

Reply:
Actually, we can't.
`append` can only use either a non-variadic or a variadic parameter, at a time.
So, combining them:
```go
 args = append(args, key, path, values...)
```
will pops error.